pkg/types/baremetal/validation/platform_test.go

```diff
@@ -0,0 +1,195 @@
+package validation
+
+import (
+	"net"
+	"testing"
+
+	"github.com/openshift/installer/pkg/ipnet"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidatePlatform(t *testing.T) {
+	iface, _ := net.Interfaces()
+	network := &types.Networking{MachineCIDR: ipnet.MustParseCIDR("192.168.111.0/24")}
+	cases := []struct {
+		name     string
+		platform *baremetal.Platform
+		network  *types.Networking
+		valid    bool
+	}{
+		{
+			name: "valid",
+			platform: &baremetal.Platform{
+				APIVIP:                  "192.168.111.2",
+				DNSVIP:                  "192.168.111.3",
+				IngressVIP:              "192.168.111.4",
+				Hosts:                   []*baremetal.Host{},
+				LibvirtURI:              "qemu://system",
+				ClusterProvisioningIP:   "172.22.0.3",
+				BootstrapProvisioningIP: "172.22.0.2",
+				ExternalBridge:          iface[0].Name,
+				ProvisioningBridge:      iface[0].Name,
+			},
+			network: network,
+			valid:   true,
```

Comment:
Instead of a `valid` bool, can we use a regexp for the expected error like [this][1]?  That ensures the error we measure is the one we expect, and not some earlier failure.

[1]: https://github.com/openshift/installer/blob/9d6030ad9fde95a82a1b8dbc2094e88b4eab0d96/pkg/types/aws/validation/machinepool_test.go#L34

Reply:
Ah nice, I didn't realize that was an option, will do - thanks!

Reply:
Done